airflow/www/views.py

```diff
@@ -612,9 +612,12 @@ def add_user_permissions_to_dag(sender, template, context, **extra):
 
         dag.can_edit = current_app.appbuilder.sm.can_edit_dag(dag.dag_id)
         dag.can_trigger = dag.can_edit and can_create_dag_run
-        dag.can_delete = current_app.appbuilder.sm.has_access(
-            permissions.ACTION_CAN_DELETE,
-            permissions.RESOURCE_DAG,
+        dag.can_delete = (
+            current_app.appbuilder.sm.has_access(
+                permissions.ACTION_CAN_DELETE,
+                permissions.RESOURCE_DAG,
+            )
+            or current_app.appbuilder.sm.can_delete_dag(dag.dag_id)
```

Comment:
```suggestion
        dag.can_delete = current_app.appbuilder.sm.can_delete_dag(dag.dag_id)
```

https://github.com/apache/airflow/blob/1739c61ad13abe9ca3388b7c5faffc6f9854fbc1/airflow/www/security.py#L328-L331

https://github.com/apache/airflow/blob/1739c61ad13abe9ca3388b7c5faffc6f9854fbc1/airflow/www/security.py#L349

https://github.com/apache/airflow/blob/1739c61ad13abe9ca3388b7c5faffc6f9854fbc1/airflow/www/security.py#L365-L368

will already take care of that